src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,164 @@
+        uint256 amount = cumulativeEigenLayerTokeEarnings[token];
+        token.safeTransfer(recipient, amount);
+        cumulativeEigenLayerTokeEarnings[token] = 0;
+    }
+
+    /**
+    * @notice Called by a staker or operator to prove the inclusion of their earnings in a posted Merkle root and claim them.
+    * @param proof Merkle proof showing that a leaf containing `(msg.sender, amount)` was included in the `rootIndex`-th
+    * Merkle root posted for the `token`
+    * @param rootIndex Specifies the node inside the Merkle tree corresponding to the specified root, `merkleRoots[rootIndex].root`.
+    */
+    function proveAndClaimEarnings(
+        bytes memory proof,
+        uint256 rootIndex,
+        MerkleLeaf memory leaf
+    ) external{
+        require(leaf.amounts.length == leaf.tokens.length, "PaymentCoordinator.proveAndClaimEarnings: leaf amounts and tokens must be same length");
+        require(merkleRoots[rootIndex].confirmedAtBlockNumber < block.number, "PaymentCoordinator.proveAndClaimEarnings: Merkle root not yet confirmed");
+        bytes32 root = merkleRoots[rootIndex].root;
+        require(root != bytes32(0), "PaymentCoordinator.proveAndClaimEarnings: Merkle root is null");
+        bytes32 leafHash = keccak256(abi.encodePacked(leaf.recipient, keccak256(abi.encodePacked(leaf.tokens)), keccak256(abi.encodePacked(leaf.amounts)), leaf.index));
+        require(Merkle.verifyInclusionKeccak(proof, root, leafHash, leaf.index), "PaymentCoordinator.proveAndClaimEarnings: Invalid proof");
+
+        for(uint i = 0; i < leaf.amounts.length; i++) {
+            leaf.tokens[i].safeTransfer(leaf.recipient, leaf.amounts[i]);
+            cumulativeTokenAmountClaimedByRecipient[leaf.tokens[i]][leaf.recipient] += leaf.amounts[i];
+        }
+        emit PaymentClaimed(leaf);
+    }
+
+    function setRootPublisher(address _rootPublisher) external onlyOwner{
+        _setRootPublisher(_rootPublisher);
+    }
+
+    function setMerkleRootActivationDelay(uint256 _merkleRootActivationDelay) external onlyOwner{
+        _setMerkleRootActivationDelay(_merkleRootActivationDelay);
+    }
+
+    function setEigenLayerShareBIPS(uint256 _eigenlayerShareBips) external onlyOwner{
+        _setEigenLayerShareBIPS(_eigenlayerShareBips);
+    }
+
+
+    /// @notice Getter function for the length of the `merkleRoots` array
+    function merkleRootsLength() external view returns (uint256){
+        return merkleRoots.length;
+    }
+
+    function _setRootPublisher(address _rootPublisher) internal {
+        rootPublisher = _rootPublisher;
+    }
+
+    function _setMerkleRootActivationDelay(uint256 _merkleRootActivationDelay) internal {
+        merkleRootActivationDelay = _merkleRootActivationDelay;
+    }
+
+    function _setEigenLayerShareBIPS(uint256 _eigenlayerShareBips) internal {
+        require(_eigenlayerShareBips <= MAX_BIPS, "PaymentCoordinator: EigenLayer share cannot be greater than 100%");
```

Comment:
there should be a stronger guarantee than this. check the docs where we discussed this; I can't remember exactly what we agreed on

Reply:
oh yeah, I think it was 15%